lib/payment.js

```diff
@@ -17,6 +18,7 @@ const _hash = new WeakMap();
 const _signature = new WeakMap();
 const _wallet = new WeakMap();
 const _provider = new WeakMap();
+const _senderRef = new WeakMap();
```

Comment:
Since the _toJSON_ now uses `_senderRef`, maybe the private variable `_senderData` needs to be removed?

Reply:
Possibly, or we could make the "senderRef" implemented ontop of _senderData. Im a bit torn about what is the best approach here, and until we land the details in the API I think I will leave this as-is for now. (single-source-of-truth discussion on the API)